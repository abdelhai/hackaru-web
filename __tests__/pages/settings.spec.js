import Factory from '@/__tests__/__setups__/factory';
import Settings from '@/pages/settings';

describe('Settings', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(Settings);
  });

  it('render correctly', () => {
    expect(factory.shallow()).toMatchSnapshot();
  });

  describe('when change selected tab', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ paths: ['home', 'settings'] });
      wrapper.find('.tabs').vm.$emit('change', 1);
    });

    it('move to selected page', () => {
      expect(factory.options.mocks.$router.push).toHaveBeenCalledWith(
        './settings'
      );
    });
  });
});
