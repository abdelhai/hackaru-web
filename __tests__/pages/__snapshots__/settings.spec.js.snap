// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings render correctly 1`] = `
<section>
  <content-header-stub>
    <tabs-stub items="account,applications,webhooks,licenses" index="0" class="tabs"></tabs-stub>
  </content-header-stub>
  <!---->
</section>
`;
